code/game/objects/structures/platforms.dm

```diff
@@ -1,156 +1,175 @@
+	if(!ismob(arrived))
+		return
+	if(arrived.throwing)
 		return NONE
-	if(!density || !(flags_atom & ON_BORDER) || !(direction & dir) || (O.status_flags & INCORPOREAL))
+	if((arrived.status_flags & INCORPOREAL))
+		return
+	if(!isturf(old_loc))
+		return
+	if(get_dist(src, old_loc) != 1) //teleporter shenanigans
+		return
+
+	var/mob/arriving_mob = arrived
+	var/old_loc_dir = get_dir(src, old_loc)
+	if(old_loc_dir & dir)
+		arriving_mob.client.move_delay += climb_slowdown //applied directly to client movedelay as mob delay would only apply after an additional movement
+
+///Applies slowdown when exiting if applicable
+/obj/structure/platform/proc/on_exit(datum/source, atom/movable/exiting, direction, list/knownblockers)
+	SIGNAL_HANDLER
+	if(!ismob(exiting))
+		return
+	if(exiting.throwing)
 		return NONE
-	knownblockers += src
-	return COMPONENT_ATOM_BLOCK_EXIT
+	if((exiting.status_flags & INCORPOREAL))
+		return
 
-/obj/structure/platform/CanAllowThrough(atom/movable/mover, turf/target)
-	. = ..()
-	if(mover?.throwing)
-		return TRUE
+	var/mob/exiting_mob = exiting
+	if(direction & dir)
+		exiting_mob.client.move_delay += climb_slowdown
 
-	var/obj/structure/S = locate(/obj/structure) in get_turf(mover)
-	if(S?.climbable && !(S.flags_atom & ON_BORDER) && climbable && isliving(mover)) //Climbable objects allow you to universally climb over others
-		return TRUE
+/obj/structure/platform/rockcliff
+	icon_state = "rockcliff"
+	name = "rock cliff"
+	desc = "A collection of stones and rocks that form a steep cliff, it looks climbable."
 
-	if(!(flags_atom & ON_BORDER) || !(get_dir(loc, target) & dir))
-		return TRUE
+/obj/structure/platform/rockcliff/icycliff
+	icon_state = "icerock"
+
+/obj/structure/platform/metalplatform
+	icon_state = "metalplatform"
 
+/obj/structure/platform/trench
+	icon_state = "platformtrench"
+	name = "trench wall"
+	desc = "A group of roughly cut planks forming the side of a dug in trench."
+
+//decorative corner platform bits
 /obj/structure/platform_decoration
 	name = "platform"
 	desc = "A square metal surface resting on four legs."
 	icon = 'icons/obj/structures/platforms.dmi'
 	icon_state = "platform_deco"
-	anchored = TRUE
-	density = FALSE
-	layer = 3.5
 	flags_atom = ON_BORDER
-	resistance_flags = UNACIDABLE
+	resistance_flags = RESIST_ALL
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
 
 /obj/structure/platform_decoration/Initialize(mapload)
 	. = ..()
 	switch(dir)
-		if (NORTH)
-			layer = ABOVE_MOB_LAYER
-		if (SOUTH)
-			layer = ABOVE_MOB_LAYER
-		if (SOUTHEAST)
-			layer = ABOVE_MOB_LAYER
-		if (SOUTHWEST)
-			layer = ABOVE_MOB_LAYER
-
-/obj/structure/platform/rockcliff
-	icon_state = "rockcliff"
-	name = "rock cliff"
-	desc = "A collection of stones and rocks that form a steep cliff, it looks climbable."
+		if(NORTH)
+			layer = ABOVE_MOB_PLATFORM_LAYER
+		if(SOUTH)
+			layer = ABOVE_MOB_PLATFORM_LAYER
+		if(SOUTHEAST)
+			layer = ABOVE_MOB_PLATFORM_LAYER
+		if(SOUTHWEST)
+			layer = ABOVE_MOB_PLATFORM_LAYER
```

Comment:
Shouldn't north draw under the mob?

Reply:
They are the two north corner bits, so the draw above the mob like the north facing platform they'd link to.